addons/note/models/note.py

```diff
@@ -0,0 +1,124 @@
+        if not vals.get('stage_id'):
+            vals['stage_id'] = self.env.context.get('default_stage_id')
+        return super(Note, self).create(vals)
+
+    @api.multi
+    def action_inactive(self):
+        self.write({'open_active': False, 'date_done': fields.date.today()})
+
+    @api.multi
+    def action_active(self):
+        self.write({'open_active': True})
+
+    #writing method (no modification of values)
+    @api.model
+    def name_create(self, name):
+        note = self.create({'memo': name})
+        return note.name_get()[0]
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        if groupby and groupby[0] == "stage_id":
+            #search all stages
+            stages = self.env['note.stage'].search([('user_id', '=', self.env.uid)])
+
+            if stages:  # if the user has some stages
+                result = [{  # notes by stage for stages user
+                    '__context': {'group_by': groupby[1:]},
+                    '__domain': domain + [('stage_ids.id', '=', stage.id)],
+                    'stage_id': (stage.id, stage.name),
+                    'stage_id_count': self.search_count(domain + [('stage_ids', '=', stage.id)]),
+                    '__fold': stage.fold,
+                } for stage in stages]
+
+                #note without user's stage
+                nb_notes_ws = self.search_count(domain + [('stage_ids', 'not in', stages.ids)])
+                if nb_notes_ws:
+                    # add note to the first column if it's the first stage
+                    dom_not_in = ('stage_ids', 'not in', stages.ids)
+                    if result and result[0]['stage_id'][0] == stages[0].id:
+                        dom_in = result[0]['__domain'].pop()
+                        result[0]['__domain'] = domain + ['|', dom_in, dom_not_in]
+                        result[0]['stage_id_count'] += nb_notes_ws
+                    else:
+                        # add the first stage column
+                        result = [{
+                            '__context': {'group_by': groupby[1:]},
+                            '__domain': domain + [dom_not_in],
+                            'stage_id': (stages[0].id, stages[0].name),
+                            'stage_id_count': nb_notes_ws,
+                            '__fold': stages[0].name,
+                        }] + result
+            else:  # if stage_ids is empty
+                #note without user's stage
+                nb_notes_ws = self.search_count(domain)
+                if nb_notes_ws:
+                    result = [{  # notes for unknown stage
+                        '__context': {'group_by': groupby[1:]},
+                        '__domain': domain,
+                        'stage_id': False,
+                        'stage_id_count': nb_notes_ws
+                    }]
+                else:
+                    result = []
+            return result
```

Comment:
where is super call for read_group  ? what if no groupby[0] == "stage_id" ?
keep this back
 return super(Note, self).read_group(domain, fields, groupby,
                offset=offset, limit=limit, orderby=orderby, lazy=lazy)